src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/EnvironmentContentTests.cs

```diff
@@ -0,0 +1,71 @@
+				Assert.IsTrue (b.Build (proj), "Build should have succeeded.");
+				var intermediateOutputDir = Path.Combine (Root, b.ProjectDirectory, proj.IntermediateOutputPath);
+				// AndroidEnableSGenConcurrent=False by default
+				List<string> envFiles = EnvironmentHelper.GatherEnvironmentFiles (intermediateOutputDir, supportedAbis, true);
+				Dictionary<string, string> envvars = EnvironmentHelper.ReadEnvironmentVariables (envFiles);
+				Assert.IsTrue (envvars.ContainsKey (gcVarName), $"Environment should contain '{gcVarName}'.");
+				Assert.AreEqual (expectedDefaultValue, envvars[gcVarName], $"'{gcVarName}' should have been '{expectedDefaultValue}' when concurrent GC is disabled.");
+
+				proj.SetProperty ("AndroidEnableSGenConcurrent", "True");
+				Assert.IsTrue (b.Build (proj), "Second build should have succeeded.");
+				envFiles = EnvironmentHelper.GatherEnvironmentFiles (intermediateOutputDir, supportedAbis, true);
+				envvars = EnvironmentHelper.ReadEnvironmentVariables (envFiles);
+				Assert.IsTrue (envvars.ContainsKey (gcVarName), $"Environment should contain '{gcVarName}'.");
+				Assert.AreEqual (expectedUpdatedValue, envvars[gcVarName], $"'{gcVarName}' should have been '{expectedUpdatedValue}' when concurrent GC is enabled.");
+			}
+		}
+
+		[Test]
+		public void CheckHttpClientHandlerType ()
+		{
+			var proj = new XamarinAndroidApplicationProject () {
+				IsRelease = true,
+			};
+			var httpClientHandlerVarName = "XA_HTTP_CLIENT_HANDLER_TYPE";
+			var expectedDefaultValue = "System.Net.Http.HttpClientHandler, System.Net.Http";
+			var expectedUpdatedValue = "Xamarin.Android.Net.AndroidClientHandler";
+			var supportedAbis = "armeabi-v7a;arm64-v8a";
+			proj.SetProperty (KnownProperties.AndroidSupportedAbis, supportedAbis);
+
+			using (var b = CreateDllBuilder (Path.Combine ("temp", TestName))) {
+				Assert.IsTrue (b.Build (proj), "Build should have succeeded.");
+				var intermediateOutputDir = Path.Combine (Root, b.ProjectDirectory, proj.IntermediateOutputPath);
+				List<string> envFiles = EnvironmentHelper.GatherEnvironmentFiles (intermediateOutputDir, supportedAbis, true);
+				Dictionary<string, string> envvars = EnvironmentHelper.ReadEnvironmentVariables (envFiles);
+				Assert.IsTrue (envvars.ContainsKey (httpClientHandlerVarName), $"Environment should contain '{httpClientHandlerVarName}'.");
+				Assert.AreEqual (expectedDefaultValue, envvars[httpClientHandlerVarName]);
+
+				proj.SetProperty ("AndroidHttpClientHandlerType", expectedUpdatedValue);
+				Assert.IsTrue (b.Build (proj), "Second build should have succeeded.");
+				envFiles = EnvironmentHelper.GatherEnvironmentFiles (intermediateOutputDir, supportedAbis, true);
+				envvars = EnvironmentHelper.ReadEnvironmentVariables (envFiles);
+				Assert.IsTrue (envvars.ContainsKey (httpClientHandlerVarName), $"Environment should contain '{httpClientHandlerVarName}'.");
+				Assert.AreEqual (expectedUpdatedValue, envvars[httpClientHandlerVarName]);
+			}
+		}
```

Comment:
Since you are adding a new test class, can you move a couple relevant tests from `BuildTest.cs` in here, too?

https://github.com/xamarin/xamarin-android/blob/cf6d51e4531ace51188d7920e2be1ef47ad9197c/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BuildTest.cs#L2045-L2136

(there might be others similar)

`BuildTest.cs` is an abomination, as we go I think we should move files out of there where we can.

Reply:
This is definitely something that's going to be needed in the future.  In my parallelization efforts of this step, the `BuildTest` class takes ~52 minutes to run, while all other test classes take ~40 minutes to run.

Reply:
Sure, I thought about this but wanted to keep the scope smaller. I'll move some things over, I definitely agree we need to work on splitting that thing up.

Reply:
Yeah I think if this PR just moved a handful of tests that are related to env vars that would be good 👍

Reply:
I moved everything that uses the `EnvironmentHelper` class into this new test file.